src/map.cpp

```diff
@@ -4727,10 +4755,18 @@ void map::draw(WINDOW* w, const point center)
     else
      mvwputch(w, realy+getmaxy(w)/2 - center.y, realx+getmaxx(w)/2 - center.x, c_ltgray, '#');
    } else if (dist <= u_clairvoyance || can_see) {
-    drawsq(w, g->u, realx, realy, false, true, center.x, center.y,
+    drawsq(w, g->u, p, false, true, center.x, center.y,
            (dist > low_sight_range && LL_LIT > lit) ||
            (dist > sight_range && LL_LOW == lit),
            LL_BRIGHT == lit);
+
+    // mark as seen
+    set_pl_known(p);
+
+   } else if (pl_knows(p)) {
+       // draw tiles that the player has seen previously but doesn't see right now
+       // as unlit and hide the items
+       drawsq(w, g->u, p, false, false, center.x, center.y, true, false);
```

Comment:
Btw. if somebody or something changes those unseen tiles, the new content will be shown to the player - magic!